### PR TITLE
Modular Components and Clubs

### PR DESCRIPTION
**Pull request by ryansherring**
# Making requested fixes to the previous PR
-modular components
-clubs refactor
-club reflects modular components
-minor naming changes and other fixes

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -3459,15 +3459,6 @@
       "resolved": "https://registry.npmjs.org/binary-extensions/-/binary-extensions-2.1.0.tgz",
       "integrity": "sha512-1Yj8h9Q+QDF5FzhMs/c9+6UntbD5MkRfRwac8DoEm9ZfUBZ7tZ55YcGVAzEe4bXsdQHEk+s9S5wsOKVdZrw0tQ=="
     },
-    "bindings": {
```
Comment (mattlaguardia) [inline package-lock.json:3462]: can we get this file out of the PR?

File: src/components/ContactInfo/ContactInfo.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+
+const ContactInfo = ({ body, title }) => {
+  let contact = body;
+
+  return (
+    <div className="contact_info">
```
Comment (mattlaguardia) [inline src/components/ContactInfo/ContactInfo.js:7]: This is the more consistent way we are doing style names, please update accordingly

```suggestion
    <div className="contact-info">
```

File: src/components/HeadingTitle/HeadingTitle.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import React from "react";
+
+const HeadingTitle = ({ rendered }) => {
+  return (
+    <h2 className="heading-title">{rendered}</h2>
+  );
+};
```
Comment (mattlaguardia) [inline src/components/HeadingTitle/HeadingTitle.js:7]: this seems a bit overkill we can just use an h2 tag no?

File: src/components/Image/Image.js [hunk 4]
```diff
@@ -2,9 +2,9 @@ import React from 'react';
 
 const Image = ({ className, webp, img, alt }) => {
   return (
-    <picture>
+    <picture className="picture">
```
Comment (mattlaguardia) [inline src/components/Image/Image.js:5]: the component is Image, so we should match the classname oh lets also not add these styles, its going break images throughout the app

File: src/components/Title/Title.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
```
Comment (mattlaguardia) [inline src/components/Title/Title.js:1]: Overkill

File: package-lock.json (hunk 1 above)
Comment (ryansherring) [reply package-lock.json:3462]: no problem!

File: src/components/Title/Title.js (hunk 5 above)
Comment (ryansherring) [reply src/components/Title/Title.js:1]: doesn't work without the React import. Should i rather just import from a single location like the other imports?

**Comment by ryansherring** [comment]
Jens changes:
1. Mobile styling for Contact Us
2. footer links
3. events page line spacing

File: src/components/Title/Title.js (hunk 5 above)
Comment (mattlaguardia) [reply src/components/Title/Title.js:1]: I dont think this has been addressed - we need to delete this file and just use h2 tags somewhere else

File: src/components/Image/Image.scss [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+    max-height: 90px;
+    margin-top: 32px;
+    margin-right: 18px;
+
+    @media screen and (max-width: $breakpoint-mobile) {
+        display: flex;
+        flex-wrap: wrap;
+        margin: auto;
+    }
+}
```
Comment (mattlaguardia) [inline src/components/Image/Image.scss:14]: Remove this - the image component shouldnt have any styles

File: src/components/components.scss [hunk 7]
```diff
@@ -7,10 +7,16 @@
   'Cards/CardEvent/CardEvent',
   'Container/Container',
   'Footer/Footer',
+  'Image/Image',
```
Comment (mattlaguardia) [inline src/components/components.scss:10]: no image stylings

File: src/components/HeadingTitle/HeadingTitle.js [hunk 8]
```diff
@@ -0,0 +1,8 @@
+import React from "react";
+
+const HeadingTitle = ({ rendered }) => {
+  return (
+    <h2>{rendered}</h2>
+  );
+};
+export default HeadingTitle;
```
Comment (mattlaguardia) [inline src/components/HeadingTitle/HeadingTitle.js:8]: No need for this component

File: src/components/Description/Description.js [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+
+const Description = ({ parent }) => {
+  return (
+    <div
+      className="description"
+      dangerouslySetInnerHTML={{ __html: parent }}
+    />
+  );
+};
+export default Description;
```
Comment (mattlaguardia) [inline src/components/Description/Description.js:11]: No need for this modular of a component

File: src/pages/Contact/Contact.scss [hunk 10]
```diff
@@ -1,21 +1,77 @@
...
+    
+    a {
+      margin: auto;
     }
+    // &--item {
+    //   display: block;
+    //   // max-width: 30px;
+    //   // max-height: 30px;
+    //   margin: auto;
+    // }
```
Comment (mattlaguardia) [inline src/pages/Contact/Contact.scss:37]: remove this please

File: src/pages/Clubs/Clubs.scss [hunk 11]
```diff
@@ -1,14 +1,12 @@
...
-  }
-
+.clubs {
   &__items {
     padding-left: 0;
     list-style: none;
     columns: 2;
     
     @media screen and (max-width: $breakpoint-mobile) {
+      margin: 0 10% 0 10%;
```
Comment (mattlaguardia) [inline src/pages/Clubs/Clubs.scss:8]: we shouldnt use 10% here, please use a standard spacing like 1rem or 2rem

also mobile style could use a second pass, this looks a bit weird to me:
Screen Shot 2021-03-31 at 1 13 59 PM

File: src/pages/Club/Club.scss [hunk 12]
```diff
@@ -0,0 +1,37 @@
+.club {
```
Comment (mattlaguardia) [inline src/pages/Club/Club.scss:1]: these pages on mobile could use a second pass at mobile styling:
Screen Shot 2021-03-31 at 1 16 16 PM

File: src/pages/Contact/Contact.js [hunk 13]
```diff
@@ -1,28 +1,27 @@
...
+        <h2 className="contact__title">Contact Us</h2>
         <div className="contact__section">
           <div className="contact__col">
           <h3>{ data.section_1.section_title }</h3>
           <div dangerouslySetInnerHTML={{ __html: data.section_1.section_body.left }} />
+          <div className="contact__social">
+          <a href="https://YouTube.com/" target="_blank" rel="noopener noreferrer"><YouTube className='contact__social--item' /></a>
+            <a href="https://Instagram.com/" target="_blank" rel="noopener noreferrer"><Instagram className='contact__social--item' /></a>
+            <a href="https://facebook.com/" target="_blank" rel="noopener noreferrer"><Facebook className='contact__social--item' /></a>
+          </div>
```
Comment (mattlaguardia) [inline src/pages/Contact/Contact.js:22]: Spacing is pretty weird here

```suggestion
            <a href="https://YouTube.com/" target="_blank" rel="noopener noreferrer">
              <YouTube className='contact__social--item' />
            </a>
            <a href="https://Instagram.com/" target="_blank" rel="noopener noreferrer">
              <Instagram className='contact__social--item' />
            </a>
            <a href="https://facebook.com/" target="_blank" rel="noopener noreferrer">
              <Facebook className='contact__social--item' />
            </a>
```

File: src/pages/Contact/Contact.js [hunk 14]
```diff
@@ -32,6 +31,7 @@ const Contact = ({ data }) => {
           <h3>{ data.section_3.section_title }</h3>
           <div className="contact__col" dangerouslySetInnerHTML={{ __html: data.section_3.section_body.left }}></div>
           <div className="contact__col" dangerouslySetInnerHTML={{ __html: data.section_3.section_body.right }}></div>
+          <div className="contact__col"><iframe src={ data.section_1.section_body.right } className="contact__map--mobile" title="Google Maps" frameBorder="0" aria-hidden="false" tabIndex="0"></iframe></div>
```
Comment (mattlaguardia) [inline src/pages/Contact/Contact.js:34]: Please double check all the spacing in the file, i see a couple things that could make this a bit easier to read

```suggestion
          <div className="contact__col">
            <iframe src={ data.section_1.section_body.right } className="contact__map--mobile" title="Google Maps" frameBorder="0" aria-hidden="false" tabIndex="0"></iframe>
          </div>
```

File: src/components/Title/Title.scss [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+    }
+
+    &__center {
+      text-align: center;
+    }
+
+    &__right {
+      text-align: end;
+    }
+}
```
Comment (mattlaguardia) [inline src/components/Title/Title.scss:17]: Needs a line at bottom of file

```suggestion
}

```

File: src/pages/Contact/Contact.js [hunk 16]
```diff
@@ -1,37 +1,47 @@
...
+            <div dangerouslySetInnerHTML={{ __html: data.section_1.section_body.left }} />
+            <div className="contact__social">
+              <a href="https://YouTube.com/" target="_blank" rel="noopener noreferrer">
+                <YouTube className='contact__social--item' />
+              </a>
+              <a href="https://Instagram.com/" target="_blank" rel="noopener noreferrer">
+                <Instagram className='contact__social--item' />
+              </a>
+              <a href="https://facebook.com/" target="_blank" rel="noopener noreferrer">
+                <Facebook className='contact__social--item' />
```
[the remark this answers is not in the document]
Comment (mattlaguardia) [reply src/pages/Contact/Contact.js:26]: this should be added to the wordpress backend lets discuss how todo this on wednesday we can pair program on it

File: src/components/ClubContactInfo/ClubContactInfo.js [hunk 17]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+
+const ClubContactInfo = ({ body }) => {
+  let contact = body;
+
+  return (
+    <div className="contact-info">
```
Comment (marco-salcedo) [inline src/components/ClubContactInfo/ClubContactInfo.js:7]: All of these should be named club-contact-info to match component name.

File: src/components/ClubContactInfo/ClubContactInfo.scss [hunk 18]
```diff
@@ -0,0 +1,17 @@
+.contact-info {
```
Comment (marco-salcedo) [inline src/components/ClubContactInfo/ClubContactInfo.scss:1]: Should be named club-contact-info to match component name.

File: src/components/Footer/Footer.js [hunk 19]
```diff
@@ -35,9 +35,9 @@ const Footer = () => {
           <div className="footer__contact--col">Mon-Fri, 8:30 am - 5 pm Saturday & Sunday, Closed</div>
         </div>
         <div className="footer__social">
-          <a href="https://twitter.com/" target="_blank" rel="noopener noreferrer"><Twitter className='footer__social-icon' /></a>
-          <a href="https://instagram.com/" target="_blank" rel="noopener noreferrer"><Instagram className='footer__social-icon' /></a>
-          <a href="https://facebook.com/" target="_blank" rel="noopener noreferrer"><Facebook className='footer__social-icon' /></a>
+          <a href="https://www.youtube.com/c/HawaiiUnitedOkinawaAssociation" target="_blank" rel="noopener noreferrer"><YouTube className='footer__social-icon' /></a>
```
Comment (marco-salcedo) [inline src/components/Footer/Footer.js:38]: Youtube component should be on its own line.

```suggestion
          <a href="https://www.youtube.com/c/HawaiiUnitedOkinawaAssociation" target="_blank" rel="noopener noreferrer">
            <YouTube className='footer__social-icon' />
          </a>
```

File: src/components/Image/Image.js [hunk 20]
```diff
@@ -4,7 +4,7 @@ const Image = ({ className, webp, img, alt }) => {
   return (
     <picture>
       <source srcSet={ webp } type='image/webp' />
-      <img className={ className } src={ img } loading='lazy' alt={ alt ? alt : 'Huoa.org Image' } />
+      <img className={`${className}`} src={ img } loading='lazy' alt={ alt ? alt : 'Huoa.org Image' } />
```
[the remark this answers is not in the document]
Comment (marco-salcedo) [reply src/components/Image/Image.js:7]: className is already a string, you don't need string interpolation

File: src/components/ImageTitleBodyButton/ImageTitleBodyButton.scss [hunk 21]
```diff
@@ -1,60 +1,60 @@
...
+  
+      &__body--right {
+        h4 {
+          margin-top: 24px;
+          margin-bottom: 28px;
+          font-size: 20px;
+        }
+      }
+    }
+  }
```
Comment (marco-salcedo) [inline src/components/ImageTitleBodyButton/ImageTitleBodyButton.scss:60]: Add empty line after bracket

File: src/components/Title/Title.js (hunk 5 above)
Comment (marco-salcedo) [reply src/components/Title/Title.js:1]: Remove this component and integrate to where its called.

File: src/components/TitleSubtitleBody/TitleSubtitleBody.scss [hunk 22]
```diff
@@ -1,27 +1,28 @@
...
+        line-height: 30px;
+      }
+      
+      &__title {
+        order: -1;
+      }
     }
   }
-}
+
```
Comment (marco-salcedo) [inline src/components/TitleSubtitleBody/TitleSubtitleBody.scss:28]: Add new line

File: src/pages/Club/Club.js [hunk 23]
```diff
@@ -0,0 +1,63 @@
...
+import Title from "components/Title/Title";
+import Button from "components/Button/Button";
+import SocialMediaLinks from "components/SocialMediaLinks/SocialMediaLinks";
+import Image from "components/Image/Image";
+
+const Club = ({ match }) => {
+  let club = useSelector((state) => state.clubs);
+  let currentClub = match.params.clubId;
+  let clubData = club.find((object) => object.slug === currentClub);
+  let button_text;
```
Comment (marco-salcedo) [inline src/pages/Club/Club.js:17]: This is never reassigned so it will always show "Optional Link".

File: src/pages/Club/Club.scss [hunk 24]
```diff
@@ -0,0 +1,47 @@
...
+      margin: 3rem 2rem;
+    }
+  }
+}
+
+.button {
+  @media screen and (max-width: $breakpoint-mobile) {
+    margin: 0 2rem;
+  }
+}
```
Comment (marco-salcedo) [inline src/pages/Club/Club.scss:47]: Add new line

File: src/pages/Contact/Contact.js [hunk 25]
```diff
@@ -1,37 +1,47 @@
...
       <div className="contact">
+        <h2 className="contact__title">Contact Us</h2>
         <div className="contact__section">
           <div className="contact__col">
-          <h3>{ data.section_1.section_title }</h3>
-          <div dangerouslySetInnerHTML={{ __html: data.section_1.section_body.left }} />
+            <h3>{data.section_1.section_title}</h3>
+            <div dangerouslySetInnerHTML={{ __html: data.section_1.section_body.left }} />
+            <div className="contact__social">
+              <a href="https://YouTube.com/" target="_blank" rel="noopener noreferrer">
```
Comment (marco-salcedo) [inline src/pages/Contact/Contact.js:19]: These social links just take you to youtube.com, etc. homepage. Not actual HUOA accounts

**Review by mattlaguardia — APPROVED** [review]
if jen approves im fine merging this in

**Pull request merged**